File: Novels/packages/5-mobile-ios/StoryReader/Model/Post.swift
import Foundation

// Post model - Social feed posts
struct Post: Codable, Identifiable {
    let id: String
    let userId: String
    let content: String
    let images: [String]?
    let storyId: String?
    let chapterId: String?
    let groupId: String?
    let likes: Int
    let comments: Int
    let shares: Int
    let createdAt: Date
    let updatedAt: Date
    
    enum CodingKeys: String, CodingKey {
        case id
        case userId = "user_id"
        case content
        case images
        case storyId = "story_id"
        case chapterId = "chapter_id"
        case groupId = "group_id"
        case likes
        case comments
        case shares
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }
}

File: Novels/packages/6-mobile-android/StoryReader/app/src/main/java/com/storysphere/storyreader/model/Post.kt
package com.storysphere.storyreader.model

data class Post(
    val id: String,
    val userId: String,
    val content: String,
    val images: List<String> = emptyList(),
    val likes: Int = 0,
    val comments: Int = 0,
    val shares: Int = 0,
    val createdAt: Long = System.currentTimeMillis(),
    val updatedAt: Long = System.currentTimeMillis()
)

